File: ruby/fizz_buzz.rb
#!/usr/bin/env ruby
def fizz?(num)
  num % 3 == 0 ? true : false
end
def buzz?(num)
  num % 5 == 0 ? true : false
end

def fizz_buzz(num)
  1.upto(num) do |n|
    if fizz?(n) && buzz?(n)
      puts 'FizzBuzz'
    elsif fizz?(n)
      puts 'Fizz'
    elsif buzz?(n)
      puts 'Buzz'
    else
      puts(n)
    end
  end
end

fizz_buzz(ARGV[0].to_i) if ARGV[0]

File: c/fizz_buzz.c
#include <stdio.h>
#include <stdlib.h>

int fizz(int num);
int buzz(int num);

int main(int argc, char* argv[])
{
        int num, i;
        if (argc <= 1) exit(0);
        num = atoi(argv[1]);
        for (i=1; i <= num; i++) {
                if (fizz(i) && buzz(i)) {
                        printf("FizzBuzz\n");
                } else if (fizz(i)) {
                        printf("Fizz\n");
                } else if (buzz(i)) {
                        printf("Buzz\n");
                } else {
                        printf("%d\n", i);
                }
        }
        return 0;
}

int fizz(int num)
{
        if ((num % 3) == 0) {
                return 1;
        }
        return 0;
}

int buzz(int num)
{
        if ((num % 5) == 0) {
                return 1;
        }
        return 0;
}
